gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.editobservation;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.view.ViewGroup;
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+import com.google.android.gnd.R;
+import com.google.android.gnd.databinding.AddPhotoListItemBinding;
+
+public class AddPhotoDialogAdapter extends RecyclerView.Adapter<AddPhotoDialogAdapter.ViewHolder> {
+
+  public static final int CAMERA = 1;
+  public static final int STORAGE = 2;
+
+  private static final int[][] ADD_PHOTO_LISt = {
```

Comment:
Maybe change this to `PHOTO_SOURCE_RESOURCES`. 
 
You could also create a separate inner class e.g. `class PhotoStorageResource { int actionStringResourceId, int iconResourceId, int sourceType }` and instead make this `private static final ArrayList<PhotoStringResource> = { ... }; // list of possible photo sources and their associated resources`

Reply:
Done